### DISCUSSION TITLE
Strange behaivour of a power monitoring plug. Only voltage and active power are correct. Current and reactive power are fixed.

### DISCUSSION THREAD

**Discussion by ortegafernando** [arendst/Tasmota#18302] [Q&A] [answered]
Hi, I have a GARZA NX-SM100 plug. The name and shape is like this: 

[https://templates.blakadder.com/tflag_NX-SM100.html](https://templates.blakadder.com/tflag_NX-SM100.html)

And I have used the template that is published.

After some test with VoltageSet, CurrentSet and PowerSet .... I have an strange behaviour.

I get voltage and active power perfectly, no problem.
But current, reactive power, power factor ..... is totally useless. But not because they are changing a lot and without any pattern, it is why the values are fixed. For example, current is fixed at the value that I have configured with current set.

If I change the load, active power is showing me the correct value of Watts, and voltage is working, but .... current and reactive power don't.

Has any body seen the strange behaviour in any other plug?

Thanks

**Comment by ortegafernando** [comment]
Hi, any boyd has got or has seen any case like this? Thanks

**Reply by sfromis** [reply]
One possibility is that you did not do a correct calibration. This is assuming that the template pins are correct for the hardware.

**Reply by ortegafernando** [reply]
Thanks but it is not the case. I know that it sounds strange but I have checked several times. Power & voltage are correctly shown. I dont exactly what GPIOs are in power meters, maybe my "old" plug is made with another meter IC than template's one?

**Comment by arendst** [comment, marked as answer]
I guess your template is wrong.

If you fail to see current correctly the HLWBL_SELi gpio is probably wrong. This pins toggles between voltage/current measuring.

You could try to change it to HLWBL_SEL without the "i" and see if anything changes.

If not then the gpio is wrong and you will need to find another gpio for this signal.

**Reply by ortegafernando** [reply]
Thanks I will try and return back with, hope, a solution. thanks.

**Reply by ortegafernando** [reply]
> I guess your template is wrong.
> 
> If you fail to see current correctly the HLWBL_SELi gpio is probably wrong. This pins toggles between voltage/current measuring.
> 
> You could try to change it to HLWBL_SEL without the "i" and see if anything changes.
> 
> If not then the gpio is wrong and you will need to find another gpio for this signal.

Thanks a lot, Finally it is other pin, instead GPIO16 , it is GPIO3 so in my casa, GARZA NX-SM100 is:
{"NAME":"GARZA NX-SM100","GPIO":[320,0,0,2624,0,2720,0,0,224,32,2656,321,0,0],"FLAG":0,"BASE":18}

I will make a PR in docs in case you decided to include this information.

Thanks a lot.
